### PR TITLE
OCPBUGS-30860: improve bootstrap logic to wait for at least 2 apiserver instances

### PR DESCRIPTION
**Pull request by tkashem**

### CODE REVIEW

**Comment by tkashem** [comment]
/hold

Obviously this breaks for SNO, so we need to omit this check for SNO

File: data/data/bootstrap/files/usr/local/bin/wait-for-ha-api.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+
+wait_for_ha_api() {
+    if [ "$BOOTSTRAP_INPLACE" = false ]
+    then
+	echo "Waiting for at least 2 available IP addresses for default/kubernetes endpoint object"
+	while :
```
Comment (vrutkovs) [inline data/data/bootstrap/files/usr/local/bin/wait-for-ha-api.sh:7]: Lets not do infinite loops and add `while (( attempts < max_attempts ))` instead

Comment (tkashem) [reply data/data/bootstrap/files/usr/local/bin/wait-for-ha-api.sh:7]: we have the same in the configmap creation loop, is there a global timeout of this script maybe?

Comment (vrutkovs) [reply data/data/bootstrap/files/usr/local/bin/wait-for-ha-api.sh:7]: oh, hmm, okay. The installer will cancel it after 30 mins anyway

**Comment by tkashem** [comment]
aws-ovn https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_installer/8394/pull-ci-openshift-installer-master-e2e-aws-ovn/1790051172988489728

**Comment by tkashem** [comment]
i do see the configmap being created, so it's getting past the wait loop in my PR
 {
            "apiVersion": "v1",
            "data": {
                "status": "complete"
            },
            "kind": "ConfigMap",
            "metadata": {
                "creationTimestamp": "2024-05-13T17:29:46Z",
                "name": "bootstrap",
                "namespace": "kube-system",
                "resourceVersion": "17739",
                "uid": "f7ac5558-ce19-4559-94ce-aa91aa3da8f1"
            }
        },
[https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshi[…]-aws-ovn/gather-extra/artifacts/configmaps.json](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_installer/8394/pull-ci-openshift-installer-master-e2e-aws-ovn/1790051172988489728/artifacts/e2e-aws-ovn/gather-extra/artifacts/configmaps.json)

File: data/data/bootstrap/files/usr/local/bin/wait-for-ha-api.sh [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    status=$?
+    if [[ $status -ne 0 ]]
+    then
+	echo "The following error happened while retrieving the default/kubernetes endpoint object"
+	echo "$output"
+	return 1
+    fi
+    
+    echo "Got the following addresses for the default/kubernetes endpoint object: $output"
+    count=$(echo "$output" | wc -w)
```
Comment (vrutkovs) [inline data/data/bootstrap/files/usr/local/bin/wait-for-ha-api.sh:33]:
```suggestion
    count=${#output}
```

Comment (tkashem) [reply data/data/bootstrap/files/usr/local/bin/wait-for-ha-api.sh:33]: I tried this, it gives me a much larger number

Comment (vrutkovs) [reply data/data/bootstrap/files/usr/local/bin/wait-for-ha-api.sh:33]: Oh, `${#` counts number of chars and `wc -w` is words

**Pull request merged**